### PR TITLE
Tow v1.0

### PR DESCRIPTION
**Pull request by claxn**
- Adapted all processes in alignment to v1.0 
- Enhancement of module structure
- Full documentation 
- Moved to `unittests`

### CODE REVIEW

**Comment by lforesta** [comment]
@claxn , looks good but tests don't work yet for me. There's a version conflict: `Requirement.parse('setuptools>=41.2'), {'xarray'})`
which version of xarray do you have locally? Maybe we need to fix its version in `setup.cfg`.

Also, you should add your name to the author/email in that file.

I think we should also change the name from eofunctions to openeo-processes-python.

**Comment by claxn** [comment]
@lforesta 
Thanks for the comment! My xarray version is 0.15.1. I will try to run the tests independently, so that I can resemble your setup and to make it work. 

I agree, renaming is a good idea, but I would suggest `openeo_processes`, because the python in the name is redundant (if you use the package you are using Python).

**Comment by m-mohr** [comment]
> I agree, renaming is a good idea, but I would suggest `openeo_processes`, because the python in the name is redundant (if you use the package you are using Python).

I disagree with the proposed name. `openeo_processes` is pretty confusing as there is the repository https://github.com/Open-EO/openeo-processes with the same name.

**Comment by claxn** [comment]
@m-mohr I totally agree with you, but I meant it differently. In Python you can import packages/modules like `from openeo_processes import mean`, which can be independently set from the repo name. So I would keep "openeo-processes-python" as a repo name (to distinguish it from the general openeo processes api repo), but when you install the repo or import it in Python you could then call `from openeo_processes import cos`. Do you agree? Because as soon as you work in Python, it is anyway clear that the packages you use are also in Python. If you do not agree with this, I am also okay with naming it as you suggested.

File: setup.cfg [hunk 1]
```diff
@@ -3,10 +3,10 @@
 # http://setuptools.readthedocs.io/en/latest/setuptools.html#configuring-setup-using-setup-cfg-files
 
 [metadata]
-name = eoFunctions
+name = openeo_processes
 description = Add a short description here!
-author = Luca Foresta
-author-email = luca.foresta@eodc.eu
+author = Claudio Navacchi
```
Comment (lforesta) [inline setup.cfg:8]: is it possible to specify multiple names? If not, I'd leave yours

File: tests/test_arrays.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 import unittest
 import numpy as np
-import eofunctions as eof
+import openeo_processes as eof
```
Comment (lforesta) [inline tests/test_arrays.py:8]: maybe: `import openeo_processes as oeo_proc` or similar?

**Comment by m-mohr** [comment]
I understood that, but still think it could be confusing. Is there a chance we release the process specs at some point for python? Then there would be a conflict. Also, we also release the clients with the language in the name, otherwise we should also make it openeo-client in Python, R and JS, I think.

File: tests/test_arrays.py (hunk 2 above)
Comment (claxn) [reply tests/test_arrays.py:8]: Ah sorry, this I forgot to rename...

File: setup.cfg (hunk 1 above)
Comment (claxn) [reply setup.cfg:8]: Sorry, I did not want to delete you on purpose. I will add you again. I thought only one person could be there.

**Comment by claxn** [comment]
Mmh okay, because at the moment also the Python client does it this way. The repo is called "openeo-python-client" and the package is called `openeo`. I thought the process specs are part of the API and remain in JSON, so I don't think there will be any conflict in the future.

**Comment by lforesta** [comment]
> Is there a chance we release the process specs at some point for python

hmn I think we should work with what we have at the moment, I agree with @claxn about the naming. We can update this in the future if needed.
In any case with python one can always rename (label) the package locally at import time, like so:
`import openeo_processes as my_processes`

File: tests/test_arrays.py (hunk 2 above)
Comment (claxn) [reply tests/test_arrays.py]: I have decided for `oeop`, I hope this okay too.

**Comment by m-mohr** [comment]
Okay, didn't knew about the python client name. I'm thinking about pypy releases and I think that's nothing you want to rename later.

**Comment by claxn** [comment]
@m-mohr Yes, sure, that would be bad and should be prevented. Lets discuss this in our next dev meeting. For now I would merge it and release the new version. What do you exactly define as a specification? You can generate a documentation from it, like you did with the openEO processes, but more than that would need additional effort.

**Comment by m-mohr** [comment]
What I'm thinking of is that it could be valuable to release the spec files (i.e. all JSON files in openeo-processes) as packages so that they can be used in implementations, which should be easier than making submodules as then they can be installed and updated through package managers.

**Comment by claxn** [comment]
Ah okay, now I understand and I agree. But no, this is actually part of the parser in the current setup. So all the specs under openeo-processes are currently also in a Python parser repo. Each back-end can then replace/modify specs (e.g. change parameter settings) etc. so that it works for their back-end. If you agree @m-mohr , I will complete the PR. Thanks for the input!

**Comment by m-mohr** [comment]
Could we wait with releasing this to PyPy until the next dev meeting? I think I'd like to check who actually would find released processes useful.

**Comment by claxn** [comment]
Sure, there is no PyPi release yet and it was not really planned to do so in the near future, only creating a tag/release on GitHub. So I will complete this PR and we discuss how we proceed with all the package release strategy during the next meeting.

**Pull request merged**